src/components/triggersAndActions.ts

```diff
@@ -78,55 +89,100 @@ export const loadScheduledActions = async function (
 }
 
 export async function processTrigger(payload: TriggerPayload) {
+  const { id: trigger_id, trigger, table, record_id } = payload
+
+  // Deduce template ID -- different for each triggered table
+  const templateId = await DBConnect.getTemplateId(table, record_id)
+
   // Get Actions from matching Template
-  const result = await PostgresDB.getActionPluginsByTemplate(payload.table, {
-    record_id: payload.record_id,
-    trigger: payload.trigger,
-  })
+  const result = await DBConnect.getActionsByTemplateId(templateId, trigger)
+
   // Filter out Actions that don't match the current condition
-  const actions: ActionInTemplate[] = []
+  // and separate into Sequential and Async actions
+  const actionsSequential: ActionSequential[] = []
+  const actionsAsync: ActionInTemplate[] = []
 
   for (const action of result) {
     const condition = await evaluateExpression(action.condition, {
       objects: [payload],
-      pgConnection: PostgresDB,
+      pgConnection: DBConnect,
     })
-    if (condition) actions.push(action)
+    if (condition) {
+      if (action.sequence) actionsSequential.push(action as ActionSequential)
+      else actionsAsync.push(action)
+    }
   }
 
-  // Evaluate parameters for each Action
-  for (const action of actions) {
-    for (const key in action.parameter_queries) {
-      action.parameter_queries[key] = await evaluateExpression(action.parameter_queries[key], {
-        objects: [payload],
-        pgConnection: PostgresDB,
-      })
-    }
-    // TODO - Error handling
-    // Write each Action with parameters to Action_Queue
-    await PostgresDB.addActionQueue({
-      trigger_event: payload.id,
+  for (const action of [...actionsAsync, ...actionsSequential]) {
+    // Add all actions to Action Queue
+    // TODO - better error handling
+    await DBConnect.addActionQueue({
+      trigger_event: trigger_id,
+      template_id: templateId,
+      sequence: action.sequence,
       action_code: action.code,
-      parameters: action.parameter_queries,
-      status: 'Queued',
+      trigger_payload: payload,
+      parameter_queries: action.parameter_queries,
+      parameters_evaluated: {},
+      status: action.sequence ? 'Processing' : 'Queued',
```

Comment:
would assume sequence would start at 1 ?

Reply:
It actually doesn't matter. It just orders them, it doesn't care about the actual numbers. Sequence 2,4,10,12 would be the same as  0,1,2,3.

Reply:
0 ? 'processing' :'queued' === 'queued'

Reply:
Ah, I see -- you're saying that if `action.sequence ===0` then it'll evaluate to False, whereas only null/undefined should. Good point, will fix.

Reply:
How about:
`status: typeof action.sequence === 'number' ? 'Processing' : 'Queued'`